airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py

```diff
@@ -74,7 +73,7 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                 (
                     stream_is_available,
                     reason,
-                ) = stream.availability_strategy.check_availability(stream, logger, self)
+                ) = stream.availability_strategy.check_availability_and_parsability(stream, logger, self)
```

Comment:
Re the conversation from the other day where I was talking about figuring out how to do a different availability check during `check` versus `read` - turns out that it didn't require access to the command.